state-chain/pallets/cf-lp/src/lib.rs

```diff
@@ -199,11 +199,15 @@ pub mod pallet {
 				Error::<T>::NoLiquidityRefundAddressRegistered
 			);
 
+			let refund_address =
+				LiquidityRefundAddress::<T>::get(&account_id, ForeignChain::from(asset));
```

Comment:
See the `ensure!` just above here. We always have this. We combine these and then we don't need the option (See other comment in the trait definition):

```suggestion
			let refund_address =
				LiquidityRefundAddress::<T>::get(&account_id, ForeignChain::from(asset))
					.ok_or(Error::<T>::NoLiquidityRefundAddressRegistered)?;
```